pkgs/applications/misc/swappy/default.nix

```
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub
+, meson
+, ninja
+, wayland
+, cairo
+, pango
+, gtk
+, pkgconfig
+, cmake
+, scdoc
+, libnotify
+, gio-sharp
+, glibc
+}:
+
+stdenv.mkDerivation rec {
+  name = "swappy-${version}";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "jtheoof";
+    repo = "swappy";
+    rev = "v${version}";
+    sha256 = "1dk5q1f71n5zs2xximpwi2jfdznqwvjrprxi7clqc077nqv48s1j";
+  };
+
+  nativeBuildInputs = [ glibc.dev ];
+
+  buildInputs = [ meson ninja pkgconfig cmake scdoc gio-sharp libnotify gtk pango cairo wayland];
+
```

It builds with this change:

```suggestion

  preConfigure = ''
    sed -i "s#gio/gunixoutputstream.h#gio-unix-2.0/gio/gunixoutputstream.h#" src/pixbuf.c
  '';

```

---

Or maybe there's a way to add `${glib.dev}/include/gio-unix-2.0` to the include path.

---

Just use:
```nix
  mesonFlags = [
    # TODO: https://github.com/NixOS/nixpkgs/issues/36468
    "-Dc_args=-I${glib.dev}/include/gio-unix-2.0"
  ];
```

and you can remove the `nativeBuildInputs`

---

and you can remove `gio-sharp`.